services/frontend-service/pkg/cmd/server.go

```diff
@@ -55,7 +58,50 @@ import (
 )
 
 var c config.ServerConfig
+var backendServiceId string = ""
+
+func getBackendServiceId(c config.ServerConfig, ctx context.Context) string {
+	if c.GKEBackendServiceID == "" && c.GKEBackendServiceName == "" {
+		logger.FromContext(ctx).Warn("gke environment variables are not set up correctly! missing backend_service_id or backend_service_name")
+		return ""
+	}
+
+	if c.GKEBackendServiceID != "" && c.GKEBackendServiceName != "" {
+		logger.FromContext(ctx).Warn("gke environment variables are not set up correctly! backend_service_id and backend_service_name cannot be set simultaneously")
+		return ""
+	}
+
+	if c.GKEBackendServiceID != "" {
+		return c.GKEBackendServiceID
+	} else {
+		regex, err := regexp.Compile(c.GKEBackendServiceName)
+		if err != nil {
+			logger.FromContext(ctx).Warn("Error compiling regex for backend_service_name: %v", zap.Error(err))
+			return ""
+		}
+		computeService, err := compute.NewService(ctx)
+		if err != nil {
+			logger.FromContext(ctx).Warn("Failed to create Compute Service client: %v", zap.Error(err))
+			return ""
+		}
+		backendServices, err := computeService.BackendServices.List(c.GKEProjectNumber).Do()
+		if err != nil {
+			logger.FromContext(ctx).Warn("Failed to get backend service: %v", zap.Error(err))
+			return ""
+		}
 
+		serviceId := ""
+		for _, backendService := range backendServices.Items {
+			if regex.MatchString(backendService.Name) {
+				serviceId = fmt.Sprint(backendService.Id)
+			}
+		}
+		if serviceId == "" {
+			logger.FromContext(ctx).Warn("No backend services found matching:", zap.String("pattern", c.GKEBackendServiceName))
+		}
+		return serviceId
+	}
```

Comment:
```suggestion
	}
        regex, err := regexp.Compile(c.GKEBackendServiceName)
        if err != nil {
	        logger.FromContext(ctx).Warn("Error compiling regex for backend_service_name: %v", zap.Error(err))
	        return ""
        }
        computeService, err := compute.NewService(ctx)
        if err != nil {
	        logger.FromContext(ctx).Warn("Failed to create Compute Service client: %v", zap.Error(err))
	        return ""
        }
        backendServices, err := computeService.BackendServices.List(c.GKEProjectNumber).Do()
        if err != nil {
	        logger.FromContext(ctx).Warn("Failed to get backend service: %v", zap.Error(err))
	        return ""
        }
        
        serviceId := ""
        for _, backendService := range backendServices.Items {
	        if regex.MatchString(backendService.Name) {
		        serviceId = fmt.Sprint(backendService.Id)
	        }
        }
        if serviceId == "" {
	        logger.FromContext(ctx).Warn("No backend services found matching:", zap.String("pattern", c.GKEBackendServiceName))
        }
        return serviceId
```

nitpick: the "else" can be simplifed